lib/controller.dart

```diff
@@ -189,332 +218,454 @@ void loadGeoMapsData() async {
-  switch (aggregate) {
-    case model.TimeAggregate.day:
-      durationToAdd = Duration(days: 1);
-      break;
-    case model.TimeAggregate.hour:
-      durationToAdd = Duration(hours: 1);
-      break;
-  }
-  while (currentTime.isBefore(endTime)) {
-    timeBucket[STANDARD_DATE_TIME_FORMAT.format(currentTime)] = 0;
-    currentTime = currentTime.add(durationToAdd);
-  }
+  if (interaction[key] == value) return true;
 
-  return timeBucket;
+  return false;
 }
 
 // Render methods
 void handleNavToAnalysis() {
   view.clearContentTab();
-  _selectedAnalysisTabIndex = 0;
-  _activeFilters = {};
-  _filterValues = {};
-  _comparisonFilterValues = {};
+  _analyseOptions.selectedTabIndex = 0;
+  _filters = [];
+
+  var uri = Uri.parse(html.window.location.href);
+  var queryParams = uri.queryParameters;
+  var chartOptions = convert.jsonDecode(queryParams['chartOptions'] ?? '{}');
+  _analyseOptions.updateFrom(chartOptions);
+
+  var queryTabLabel = _config.tabs[_analyseOptions.selectedTabIndex].label;
 
   var tabLabels =
       _config.tabs.asMap().map((i, t) => MapEntry(i, t.label)).values.toList();
-  view.renderAnalysisTabs(tabLabels);
-  view.renderChartOptions(_dataComparisonEnabled, _dataNormalisationEnabled,
-      _stackTimeSeriesEnabled);
-
-  _computeFilterDropdownsAndRender();
-  _computeChartBucketsAndRender();
+  view.renderAnalysisTabs(tabLabels, queryTabLabel);
+  view.renderChartOptions(
+      _analyseOptions.dataComparisonEnabled,
+      _analyseOptions.normaliseDataEnabled,
+      _analyseOptions.stackTimeseriesEnabled);
+
+  _computeFilterDropdownsAndRender(
+      _config.tabs[_analyseOptions.selectedTabIndex].filters);
+  _computeChartDataAndRender();
 }
 
-void _computeFilterDropdownsAndRender() {
-  var filterKeys = _config.filters.map((filter) => filter.key).toList();
-  var filterOptions = _uniqueFieldCategoryValues.map((key, setValues) {
-    return MapEntry(
-        key,
-        setValues.map((s) => s.toString()).toList()
-          ..add(DEFAULT_FILTER_SELECT_VALUE));
-  });
+void _computeFilterDropdownsAndRender(List<model.Filter> filters) {
+  filters = filters ?? [];
+
+  _filters = [];
+  filters.forEach((f) {
+    var filterOptions = <String>[];
+    var defaultValue = DEFAULT_FILTER_SELECT_VALUE;
+    if (f.type == model.DataType.string) {
+      filterOptions = _uniqueFieldValues[f.data_path][f.key]
+          .map((e) => e.toString())
+          .toList()
+            ..add(DEFAULT_FILTER_SELECT_VALUE);
+    } else if (f.type == model.DataType.datetime) {
+      filterOptions = [];
+      switch (f.data_path) {
+        case model.DataPath.message_stats:
+          // todo: check across all _messageStats, not just the first
+          var dates = _messageStats.values.first.keys.toList()..sort();
+          filterOptions = [dates.first, dates.last];
+          defaultValue = dates.first.toString().split('T').first +
+              '_' +
+              dates.last.toString().split('T').first;
```

Comment:
a little comment would be helpful here that for dates the default is the whole range, and that that is what you're building - otherwise it takes a little time to read this

Reply:
Ah sure, added and will keep in mind